### PR TITLE
Fix reaction_func

### PR DESCRIPTION
**Pull request by isaacsas**

### CODE REVIEW

**Comment by isaacsas** [comment]
@TorkelE seems to pass tests. Look ok to you?

File: src/reaction_network.jl [hunk 1]
```diff
@@ -233,11 +233,12 @@ function recursive_expand_functions!(expr::ExprValues)
     (typeof(expr)!=Expr) && (return expr)
     foreach(i -> expr.args[i] = recursive_expand_functions!(expr.args[i]), 1:length(expr.args))
     if expr.head == :call
-        haskey(funcdict, expr.args[1]) && return funcdict[expr.args[1]](expr.args[2:end])
         in(expr.args[1],hill_name) && return hill(expr)
         in(expr.args[1],hillR_name) && return hillR(expr)
         in(expr.args[1],mm_name) && return mm(expr)
         in(expr.args[1],mmR_name) && return mmR(expr)
+
+        !isdefined(Catalyst,expr.args[1]) && (expr.args[1] = esc(expr.args[1]))
```
Comment (ChrisRackauckas) [inline src/reaction_network.jl:241]: the answer was this esc?

Comment (isaacsas) [reply src/reaction_network.jl:241]: Seems to work for the simple examples I’ve tested. Not merging yet as I’d like to try a couple more complicated expressions to double check tomorrow, including your NN example.

**Comment by TorkelE** [comment]
Looks really neat!

**Comment by TorkelE** [comment]
(not at a computer with Julia right now, but will try a few tests tomorrow if I find time)

**Comment by isaacsas** [comment]
Seems to close https://github.com/SciML/Catalyst.jl/issues/248 and https://github.com/SciML/Catalyst.jl/issues/276 as far as getting a `ReactionSystem` generated. We should next test system conversion and update MT if needed.

I'm going to close those issues since the Catalyst end seems ok now and merge this.

**Pull request merged**